cli/src/invocation/mod.rs

```diff
@@ -100,35 +100,33 @@ pub(crate) fn invoke(command: Command, settings: Config) -> anyhow::Result<()> {
 // utilities for invoke
 
 /// Get the replica for this invocation
-fn get_replica(settings: &Config) -> anyhow::Result<Replica> {
-    let taskdb_dir = settings.get_str("data_dir")?.into();
+fn get_replica(settings: &Settings) -> anyhow::Result<Replica> {
+    let taskdb_dir = settings.data_dir.clone();
     log::debug!("Replica data_dir: {:?}", taskdb_dir);
     let storage_config = StorageConfig::OnDisk { taskdb_dir };
     Ok(Replica::new(storage_config.into_storage()?))
 }
 
 /// Get the server for this invocation
-fn get_server(settings: &Config) -> anyhow::Result<Box<dyn Server>> {
+fn get_server(settings: &Settings) -> anyhow::Result<Box<dyn Server>> {
     // if server_client_key and server_origin are both set, use
     // the remote server
-    let config = if let (Ok(client_key), Ok(origin)) = (
-        settings.get_str("server_client_key"),
-        settings.get_str("server_origin"),
+    let config = if let (Some(client_key), Some(origin), Some(encryption_secret)) = (
+        settings.server_client_key.as_ref(),
+        settings.server_origin.as_ref(),
+        settings.encryption_secret.as_ref(),
```

Comment:
Is it an issue that previously the encryption key missing was an error if the client key+origin were set, but now it will quietly ignore the remote server?

I guess I would expect that if the `server_origin` is set, the client key and encryption secret are both required (and it's an error to omit them)

Probably something to split into a new issue and worry about later, but I wonder if it'd make sense to have the entire server address as a single string (along the lines of the HTTP `key:secret@serveradd` or Postgres connection strings etc). Would be easier for users to copy one string, and nicer for the code to handle

Reply:
I'm not too worried about the change (I suspect it was an oversight when I added encryption).  In general this "if-it's-present" is not a great UI.  But, let's solve that when the server bits are more stable.